saw-core/src/SAWCore/Term/CtxTerm.hs

```diff
@@ -88,263 +47,42 @@ import SAWCore.Term.Functor
 -- * Contexts and Bindings
 --
 
--- | A representation of the type of SAW types as a Haskell type. This is
--- actually a singleton type, meaning that a 'CtxTerm' with type @'Typ' a@ is a
--- SAW type that is represented by Haskell type @a@. Of course, the Haskell type
--- system is not rich enough to capture SAW types in complete detail, but we do
--- our best, and capture at least the types and functions.
-data Typ (a :: Type)
-
--- | An identifier for a datatype that is statically associated with Haskell
--- type @d@. Again, we cannot capture all of the SAW type system in Haskell, so
--- we simplify datatypes to arbitrary Haskell types.
-newtype DataIdent d = DataIdent Name
-  -- Invariant, the type of datatypes is always a closed term
-
--- | Append a list of types to a context, i.e., "invert" the list of types,
--- putting the last type on the "outside", and append it. The way to think of
--- this operation is that we are already "inside" @ctx@, and we are moving
--- further "inside" of @as@, one type at a time, to yield a combined context
--- where the last type of @as@ is bound last, i.e., most recently.
-type family CtxInvApp ctx as where
-  CtxInvApp ctx '[] = ctx
-  CtxInvApp ctx (a ': as) = CtxInvApp (ctx ::> a) as
-
--- | Invert a type list to make a context
-type CtxInv as = CtxInvApp EmptyCtx as
-
--- | A sequence of bindings of pairs of a variable name and a type of some form
--- for that variable. These bindings are relative to ambient context @ctx@, use
--- @tp@ for the variable types, and bind variables whose types are listed in
--- @as@.
---
--- Note that each type in a bindings list has type 'Typ', but is "represented"
--- by a Haskell type @a@ in the 'Bind' constructor. There is no way to actually
--- related the Haskell type @a@ to the type it "represents", so we do not try,
--- and just write "represent" in quotes.
-data Bindings (tp :: Ctx Type -> Type -> Type) (ctx :: Ctx Type) (as :: [Type]) where
-  NoBind :: Bindings tp ctx '[]
-  Bind :: LocalName -> tp ctx (Typ a) -> Bindings tp (ctx ::> a) as ->
-          Bindings tp ctx (a ': as)
-
--- | Compute the number of bindings in a bindings list
-bindingsLength :: Bindings tp ctx as -> Int
-bindingsLength NoBind = 0
-bindingsLength (Bind _ _ bs) = 1 + bindingsLength bs
-
--- | An inverted list of bindings, seen from the "inside out"
-data InvBindings (tp :: Ctx Type -> Type -> Type) (ctx :: Ctx Type) (as :: Ctx Type) where
-  InvNoBind :: InvBindings tp ctx EmptyCtx
-  InvBind :: InvBindings tp ctx as -> LocalName -> tp (ctx <+> as) (Typ a) ->
-             InvBindings tp ctx (as ::> a)
-
--- | Compute the number of bindings in an inverted bindings list
-invBindingsLength :: InvBindings tp ctx as -> Int
-invBindingsLength InvNoBind = 0
-invBindingsLength (InvBind bs _ _) = 1 + invBindingsLength bs
-
 -- | Map over all types in an inverted bindings list
```

Comment:
Now that we are representing bindings as an ordinary Haskell list, in what sense are they "inverted"?

Reply:
The "inverted" stuff was removed in 2fec3237ffa380956263edc3db491c662d04d471 and 605259ed5bc577415a44bf0183be31f8832b465c.